kratos/linear_solvers/linear_solver.h

```diff
@@ -273,6 +273,12 @@ class LinearSolver
         return 0;
     }
 
+    virtual unsigned int GetIterationsNumber()
+    {
+        std::cout << "WARNING: Accessed base function Kratos::LinearSolver::GetIterationsNumber(). Iterations number is only defined for AMGCL and IterativeSolver." << std::endl;
```

Comment:
```suggestion
        KRATOS_WARNING("LinearSolver") << "Accessed base function LinearSolver::GetIterationsNumber." << std::endl;
```

please don't use `std::cout` directly